Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+                    OandaEnvironment = Environment.Sandbox;
+                    break;
+                case "practice":
+                    OandaEnvironment = Environment.Practice;
+                    break;
+                case "trade":
+                    OandaEnvironment = Environment.Trade;
+                    break;
+                default:
+                    throw new ArgumentException("Unexpected or unexpected Oanda Environment: " + environment);
+            }
+
+        }
+        
+        /// <summary>
+        /// Converts the specified tradier order into a qc order.
+        /// The 'task' will have a value if we needed to issue a rest call for the stop price, otherwise it will be null
+        /// </summary>
+        protected Order ConvertOrder(DataType.Order order)
+        {
+            Order qcOrder;
+            switch (order.type)
+            {
+                case "limit":
+                    qcOrder = new LimitOrder();
+                    if (order.side == "buy")
+                    {
+                        ((LimitOrder)qcOrder).LimitPrice = Convert.ToDecimal(order.lowerBound);
+                    }
+
+                    if (order.side == "sell")
+                    {
+                        ((LimitOrder)qcOrder).LimitPrice = Convert.ToDecimal(order.upperBound);
+                    }
+
+                    break;
+                case "stop":
+                    qcOrder = new StopLimitOrder();
+                    if (order.side == "buy")
+                    {
+                        ((StopLimitOrder)qcOrder).LimitPrice = Convert.ToDecimal(order.lowerBound);
+                    }
+
+                    if (order.side == "sell")
+                    {
+                        ((StopLimitOrder)qcOrder).LimitPrice = Convert.ToDecimal(order.upperBound);
+                    }
+                    break;
+                case "marketIfTouched":
+                    //when market reaches the price sell at market.
+                    qcOrder = new StopMarketOrder { Price = Convert.ToDecimal(order.price), StopPrice = Convert.ToDecimal(order.price)};
+                    break;
+                case "market":
+                    qcOrder = new MarketOrder();
+                    break;
+
+                default:
+                    throw new NotSupportedException("The Oanda order type " + order.type + " is not supported.");
+            }
+            qcOrder.Symbol = order.instrument;
+            qcOrder.Quantity = ConvertQuantity(order);
+            qcOrder.SecurityType = InstrumentSecurityTypeMap[order.instrument];
+            qcOrder.Status = OrderStatus.None;
+            qcOrder.BrokerId.Add(order.id);
+            qcOrder.Id = order.id;
+            qcOrder.Duration = OrderDuration.Specific;
+            qcOrder.DurationValue = XmlConvert.ToDateTime(order.expiry, XmlDateTimeSerializationMode.Utc);
+            qcOrder.Time = XmlConvert.ToDateTime(order.time, XmlDateTimeSerializationMode.Utc);
```

Comment:
great stuff, we'll need to discuss supporting this new DurationValue in backtesting, currently we only really support good til cancelled